src/pczt/tx_extractor.rs

```diff
@@ -126,24 +128,29 @@ impl Authorization for Unbound {
     type SpendAuth = redpallas::Signature<SpendAuth>;
 }
 
-impl<V> crate::Bundle<Unbound, V> {
+impl<V> crate::Bundle<Unbound, V, OrchardVanilla> {
     /// Verifies the given sighash with every `spend_auth_sig`, and then binds the bundle.
     ///
     /// Returns `None` if the given sighash does not validate against every `spend_auth_sig`.
     pub fn apply_binding_signature<R: RngCore + CryptoRng>(
         self,
         sighash: [u8; 32],
         rng: R,
-    ) -> Option<crate::Bundle<Authorized, V>> {
+    ) -> Option<crate::Bundle<Authorized, V, OrchardVanilla>> {
         if self
             .actions()
             .iter()
             .all(|action| action.rk().verify(&sighash, action.authorization()).is_ok())
         {
             Some(self.map_authorization(
                 &mut (),
-                |_, _, a| a,
-                |_, Unbound { proof, bsk }| Authorized::from_parts(proof, bsk.sign(rng, &sighash)),
+                |_, _, a| VerSpendAuthSig::new(OrchardSighashVersion::V0, a),
+                |_, Unbound { proof, bsk }| {
+                    Authorized::from_parts(
+                        proof,
+                        VerBindingSig::new(OrchardSighashVersion::V0, bsk.sign(rng, &sighash)),
```

Comment:
Non-blocking: The sighash version needs to be passed into the method, because it is a property of the sighash (and only the caller knows which version was used). Given that this will need an API refactor across several crates to ensure consistency, let's not do it in this PR (which is complicated enough as it is).